libsolidity/codegen/ir/IRGenerationContext.h

```diff
@@ -43,6 +44,26 @@ class FunctionDefinition;
 class Expression;
 class YulUtilFunctions;
 
+/**
+ * A very thin wrapper over a collection of function definitions.
+ * It provides a queue-like interface but does not guarantee element order and does not
+ * preserve duplicates.
+ */
+class IRFunctionGenerationQueue
```

Comment:
Does this class provide such a big benefit?

The only non-trivial function - `pop` - is called from exactly one place.

Reply:
I created it only to avoid adding a bunch of loose functions with long names directly into the context. A class groups it nicely. Would you prefer it in the context instead?